extras/linter_install_tutorial.md

```diff
@@ -0,0 +1,96 @@
+# Install Linter to Automatically Run Before Git Commit
+
+## About
+
+This section describes how to set up running a liner uploading code onto a GitHub repository (git commit ... -> git push).
+A [linter](https://en.wikipedia.org/wiki/Lint_(software)) is a code analysis tool designed to analyze code and flag for programming errors, bugs and stylistic errors.
+This tool allows one to drastically improve their productivity when writing code for their research projects as stylistic errors and programming errors will automatically be covered through this set-up.
+Disclaimer: The following instructions only work for python code as R doesn't have a working linter. (Subject to change if new information were to come out on this fact.)
+
+## Programs and Packages needed to set up Pipeline
+
+Make sure [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/download.html) or another python package manager is installed on your computer.
+This pipeline is designed to run on python >=3.5.
+
+This setup will also use the following packages (versions can be changed as needed):
+1. pre-commit==2.9.3
+2. black==20.8b1 
+3. jupytext==1.9.1
+
+## Installation Instructions
+
+1. Set up your python environment to use python>=3.5 
+2. Call ``pip install pre-commit==2.9.3``
+3. call ``pip install black==20.8b1``
+4. Create a file with the extension .pre-commit-config.yaml and fill in with lines of code reproduced in the [Configuration Files](#Pre-commit) section
```

Comment:
Not sure if this is needed but the extension name needs to exactly match. `yml` instead of `yaml` didn't work

Reply:
Sorry what you had was correct (`.yaml`). My comment was to say that when I initially read the instructions I didn't think the extension needed to be exact and so I named my file `.pre-commit-config.yml` and it did not work. But using `.pre-commit-config.yaml` did work

Reply:
This comment was more to say that I'm not sure if others will read it the same way I did initially and was wondering if we needed to add a sentence to say "these extension names need to be exactly ____"? But I think this was something that I easily figured out once it didn't work so its not a big deal to leave the instructions as is I think. So sorry for the confusion